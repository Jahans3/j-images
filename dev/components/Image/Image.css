.listItem {
    display: inline-block;
}

.innerWrap {
    height: 300px;
    width: 300px;
}