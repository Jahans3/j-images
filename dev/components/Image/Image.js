/**
 * Created by jahansj on 19/11/2016.
 */
import React, { Component } from 'react';
import s from './Image.css';

export default class Image extends Component {
  constructor() {
    super();

    this.extractID = this.extractID.bind(this);
    this.toggleSelected = this.toggleSelected.bind(this);

    this.state = {
      selected: {
        isSelected: false,
        className: 'selected',
        id: this.extractID()
      }
    };
  }

  extractID() {
    if (!this.props) return;

    let src = this.props.src;

    src = src.split('/');
    src = src[src.length - 1];

    return src.split('.')[0];
  }

  toggleSelected() {
    const ls = window.localStorage;
    const existing = JSON.parse(localStorage.getItem('flickrImages')) || [];
    const index = existing.indexOf(`imageK${this.props.idKey}`);

    if (index !== -1) {
      existing.splice(index, 1);
      this.setState({ isSelected: false });
    } else {
      existing.push(`imageK${this.props.idKey}`);
      this.setState({ isSelected: true });
    }

    console.log('toggley toggley');
    console.log(ls.getItem('flickrImages'));

    ls.setItem('flickrImages', JSON.stringify(existing));
  }
  
  componentWillMount() {
    // if ID exists in localStorage set state.isSelected to true
    const selectedimages = window.localStorage.getItem('')
  }

  componentDidMount() {
    document.getElementById(`imageK${this.props.idKey}`).addEventListener('click', this.toggleSelected);
  }

  render() {

    return (
        <li className={ s.listItem }>
          <div id={ `imageK${this.props.idKey}` } className={ `${s.innerWrap} ${this.state.isSelected ? 'selected' : ''}` }>
            <h4 className={ s.title }>
              {
                this.props.title
              }
            </h4>
            
            <img src={ this.props.src } alt={ this.props.alt } />
          </div>
        </li>
    )
  }
}