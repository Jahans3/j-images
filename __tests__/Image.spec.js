/**
 * Created by jahansj on 19/11/2016.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../dev/components/Image/Image';

const addProps = (obj, mock, value) => obj[mock] = value;
const lsMock = {
  getItem: item => null,
  setItem: (key, value) => true
};

function setState(obj) {
  for (let prop in obj) {
    if (obj.hasOwnProperty(prop)) {
      this[prop] = obj[prop];
    }
  }
}

const obj = {};

addProps(window, 'localStorage', lsMock);
addProps(obj, 'setState', setState);
addProps(obj, 'props', {});
addProps(obj.props, 'idKey', '');

test('Image', () => {
  const tree = renderer.create(
      <Image
        title={ 'Title Text' }
        src={ 'http://some.src' }
        alt={ 'Alt Text' }
        idKey={0}
      />
  ).toJSON();
  
  expect(tree).toMatchSnapshot();
});

test('extractID', () => {
  addProps(obj, 'extractID', Image.prototype.extractID);
  addProps(obj, 'props', {});
  addProps(obj.props, 'src', 'http://some.src/path.ext');

  expect(obj.extractID()).toEqual('path');
});

test('toggleSelected', () => {
  addProps(obj, 'toggleSelected', Image.prototype.toggleSelected);
  
  spyOn(localStorage, 'setItem');
  
  obj.toggleSelected();

  expect(localStorage.setItem).toHaveBeenCalled();
});

test('findSelectedImages', () => {
  addProps(obj, 'findSelectedImages', Image.prototype.findSelectedImages);

  expect(obj.findSelectedImages()).toEqual(undefined);
});