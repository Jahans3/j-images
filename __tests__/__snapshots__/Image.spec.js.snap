exports[`test Image 1`] = `
<li
  className={undefined}>
  <div
    className="undefined "
    id="imageK0">
    <h4
      className={undefined}>
      Title Text
    </h4>
    <img
      alt="Alt Text"
      src="http://some.src" />
  </div>
</li>
`;
